VnE/Source/ShaderPipelines.py

```
@@ -395,28 +394,62 @@ def bind_id():
+        glBindProgramPipeline(pipeline)
 
         for i, program in enumerate(self.programs):
             self.programs[i][1] = SHADER_PROGRAM_CREATOR.createProgram(program[2], program[0])[0]
-            glUseProgramStages(self.pipeline, BallsShaderPipeline.shader_bit[self.programs[i][0]], self.programs[i][1])
+            glUseProgramStages(pipeline, self.shader_bit[self.programs[i][0]], self.programs[i][1])
 
         self.uniforms = {'const_scale': Uniform(np.float32, 'const_scale', self.programs[0][1])}
         self.uniforms['const_scale'].set(ctypes.c_float(150.0))
-        glUseProgram(0)
+
+        self.pipeline = self.Program(pipeline, GL_TRIANGLES)
```

Ruff (0.14.14)

[error] `GL_TRIANGLES` may be undefined, or defined from star imports

(F405)